### PR TITLE
Update server api to use new pipeline model

### PR DESCRIPTION
**Pull request by sunya-ch**
This PR extended and updated the server-api to work with new pipeline models (https://github.com/sustainable-computing-io/kepler-model-server/issues/75).

- [x] There are some dependencies in PR https://github.com/sustainable-computing-io/kepler-model-server/pull/88. 
Need to merge that PR first. 

Signed-off-by: Sunyanan Choochotkaew [sunyanan.choochotkaew1@ibm.com](mailto:sunyanan.choochotkaew1@ibm.com)

### CODE REVIEW

File: src/server/model_server.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+app = Flask(__name__)
+
+@app.route('/model', methods=['POST'])
+def get_model():
+    model_request = request.get_json()
+    req = ModelRequest(**model_request)
+    valid_fgs = get_valid_feature_groups(req.metrics)
+    filters = parse_filters(req.filter)
+    print(req.output_type)
```
Comment (KaiyiLiu1234) [inline src/server/model_server.py:78]: A few of these print statements may not be necessary and should be removed or logged for future reference.

File: src/server/model_server.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+from train import get_valid_feature_groups, is_weight_output, ModelOutputType, FeatureGroups, FeatureGroup
+from util.config import getConfig
+from util.loader import parse_filters, is_valid_model, load_json, get_model_weight, get_model_group_path, get_archived_file, METADATA_FILENAME, CHECKPOINT_FOLDERNAME
+
+###############################################
+# model request 
+
+class ModelRequest():
+    def __init__(self, metrics, output_type, source='rapl', node_type=-1, weight=False, trainer_name="", filter=""):
```
Comment (KaiyiLiu1234) [inline src/server/model_server.py:21]: For future reference, might be a good idea to include comments and class descriptors just to make it easier for new developers to onboard

Comment (sunya-ch) [reply src/server/model_server.py:21]: Thanks. Added comments for each attribute.
https://github.com/sustainable-computing-io/kepler-model-server/compare/a989d7051b8056261c5df86a5594af3704517201..04aae778b8a000ff5209aa4d7008143d2262aa96

File: src/server/model_server.py (hunk 1 above)
Comment (sunya-ch) [reply src/server/model_server.py]: Change from printing to comment. Could we put changing to logging as a new issue and PR applying to other component (such as online trainer) as well?

**Pull request merged**